app/main/posts/modify/post-data-editor.controller.js

```diff
@@ -84,20 +86,51 @@ function PostDataEditorController(
         $scope.savePost();
     });
 
-    $scope.$on('event:edit:leave:form', function () {
-        if ($scope.parentForm.form && $scope.parentForm.form.$dirty) {
-            $scope.leavePost();
-        } else {
-            $scope.cancel();
+    $transitions.onStart({}, function (transition) {
+        //where is it going? transition.to().name
+        // return rejected promise or false to cancel the transition
+        // leavePost calls cancel which then resolves or rejects the state change.
+        if (!ignoreCancelEvent && transition.from().name === 'list.data.edit') {
+            return $scope.leavePost(transition);
         }
-    });
 
-    /**
-     * @uirouter-refactor check this and add back if neded
-     * var $locationChangeStartHandler = $scope.$on('$locationChangeStart', function (e, next) {
-        $scope.leavePost(next, e);
+        return true;
+
     });
-     */
+
+    function leavePost(transition) {
```

Comment:
I think renaming this might make things clearer. ie. `checkUnsaved` or `saveAndLeave` ... I don't know but its hard to follow whats going on here.